x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/popover_editor.tsx

```diff
@@ -160,6 +160,11 @@ export function PopoverEditor(props: PopoverEditorProps) {
           compatibleWithCurrentField ? '' : ' incompatible'
         }`,
         onClick() {
+          // todo: when moving from terms agg to filters, we want to create a filter `$field.name : *`
```

Comment:
The problem here is that other columns are already accepted if the field is compatible with them (the fact they are not accepted was the misconception I got from checking the previousColumn type we talked about async). In our case, top values and filters have totally non-compatible fields set.  I think this has to be completely changed anyway once we allow creating fieldless columns. I'll create an issue, but without proposal for implementation, because it depends on fieldless column. Feel free to add some details if you have any ideas.